src/Storages/VirtualColumnUtils.cpp

```diff
@@ -132,6 +145,15 @@ VirtualColumnsDescription getVirtualsForFileLikeStorage(const ColumnsDescription
     add_virtual("_size", makeNullable(std::make_shared<DataTypeUInt64>()));
     add_virtual("_time", makeNullable(std::make_shared<DataTypeDateTime>()));
 
+    auto map = parseFromPath(path);
+    for (const auto& item : map)
+    {
+        auto type = tryInferDataTypeForSingleField(item.second, settings);
```

Comment:
Let's add tests for different types.
Something like:
`hive_partitioning/number=42/date=2020-01-01/sample.parquet`, `hive_partitioning/array=[1,2,3]/float=42.42/sample.parquet`